drud/ddev
# Custom container build cleanup, fixes #2021, fixes #1855

## The Problem/Issue/Bug:

#2021 points out that currently all projects end up using the same customized container. Nobody ever complained about this, but it's not right. This adds the project name to the base container to disambiguate.
#1855 points out that users can't add files properly from a custom Dockerfile because the context is not set correctly.

## How this PR Solves The Problem:

* Disambiguate the name
* Add new entry in docker-compose.yaml to specify context *and* Dockerfile

## Manual Testing Instructions:

* Use webimage_extra_packages in one project that is different from that in another.
* `ddev start` and `ddev ssh` and verify that packages are different.
* Add a file to .ddev/web-build/ (like "junk.txt")
* Add to the .ddev/web-build/Dockerfile `ADD junk.txt /`
* `ddev start`
* `ddev ssh` and verify that junk.txt is now in the root.

## Automated Testing Overview:


## Related Issue Link(s):

## Release/Deployment notes:
